File: June-18/04-06-18/python/01-Mirror-image-of-Mahatama-Ganghi.py
def pali(data, n):
    for i in range(0, n//2):
        if(data[i] != data[n-i-1]):
            return False
    return True
for _ in range(int(input())):
    data = input()
    n = len(data)
    flag = False
    
    for i in range(0, n):
        if(data[i] == '0' or data[i] == '1' or data[i] == '8'):
            continue
        else:
            flag = True
            break
    if(flag == False and pali(data, n) == True):
        print("YES")
    else:
        print("NO")

File: June-18/04-06-18/java/01-Mirror-image-of-Mahatama-Ganghi.java

import java.io.*;


class TestClass {
    static boolean pali(String str, int n)
    {
        for(int i = 0; i < n/2; i++)
        {
            if(str.charAt(i) != str.charAt(n-i-1))
                return false;
        }
        return true;
    }
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        while(t-- > 0)
        {
            String str = input.readLine();
            int n = str.length();
            boolean flag = false;
            for(int i = 0; i < n; i++)
            {
                char ch = str.charAt(i);
                if(ch == '0' || ch == '1' || ch == '8')
                    continue;
                else
                {
                    flag = true;
                    break;
                }
            }
            if(flag == false && pali(str, n) == true)
                System.out.println("YES");
            else
                System.out.println("NO");
            
            
        }
        
    }
}
